x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/common/io/SqlStreamOutput.java

```diff
@@ -8,35 +8,95 @@
 package org.elasticsearch.xpack.sql.common.io;
 
 import org.elasticsearch.Version;
+import org.elasticsearch.common.compress.CompressorFactory;
 import org.elasticsearch.common.io.stream.OutputStreamStreamOutput;
+import org.elasticsearch.common.io.stream.StreamOutput;
 
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
-import java.nio.charset.StandardCharsets;
+import java.io.OutputStream;
 import java.time.ZoneId;
 import java.util.Base64;
 
-public class SqlStreamOutput extends OutputStreamStreamOutput {
+/**
+ * Output stream for writing SQL cursors. The output is compressed if it would become larger than {@code compressionThreshold}
+ * bytes otherwise (see {@code DEFAULT_COMPRESSION_THRESHOLD}).
+ *
+ * The wire format is {@code version compressedFlag [payload | compressedPayload]}.
+ */
+public class SqlStreamOutput extends StreamOutput {
+
+    public static final byte HEADER_UNCOMPRESSED = 0;
+    public static final byte HEADER_COMPRESSED = 1;
```

Comment:
Since you're not using a cookie, wouldn't a boolean suffice in this case? It's also serialize to a byte.

Reply:
🤦 yes, that can be simpler indeed 👍